cmd/access/node_builder/access_node_builder.go

```diff
@@ -1605,7 +1614,7 @@ func (builder *FlowAccessNodeBuilder) Build() (cmd.Node, error) {
 			return nil
 		}).
 		Module("backend script executor", func(node *cmd.NodeConfig) error {
-			builder.ScriptExecutor = backend.NewScriptExecutor(builder.Logger, builder.scriptExecMinBlock, builder.scriptExecMaxBlock)
+			builder.ScriptExecutor = backend.NewScriptExecutor(builder.Logger, builder.versionControl, builder.scriptExecMinBlock, builder.scriptExecMaxBlock)
```

Comment:
`Module()` are run before `Component()`, so `builder.versionControl` will always be nil here

Reply:
I think the simplest way to do it is:
1. Remove version control from `NewScriptExecutor`
2. Add it to `ScriptExecutor.Initialize()`
3. Make `DependableComponent("execution data indexer")` depend on the version control component
4. Update `Component("version control")` so that it is always started, but if `builder.versionControlEnabled` is false, it returns a `module.NoopReadyDoneAware` instead of creating the version control instance.

This way, when `ScriptExecutor.Initialize()` is called, the version control is guaranteed to be setup (though possibly still nil if disabled)